crates/starknet-devnet/tests/test_blocks_on_demand.rs

```diff
@@ -55,14 +57,78 @@ mod blocks_on_demand_tests {
         }
     }
 
-    async fn assert_pending_block_with_transactions(devnet: &BackgroundDevnet) {
+    async fn assert_pending_block_with_tx_hashes(devnet: &BackgroundDevnet) {
         let pending_block = devnet.get_pending_block_with_tx_hashes().await.unwrap();
 
         for tx_hash in pending_block.transactions {
             assert_tx_successful(&tx_hash, &devnet.json_rpc_client).await;
         }
     }
 
+    async fn assert_latest_block_with_txs(devnet: &BackgroundDevnet, block_number: u64) {
+        let latest_block = devnet.get_latest_block_with_txs().await.unwrap();
+
+        assert_eq!(latest_block.block_number, block_number);
+        assert_eq!(latest_block.status, BlockStatus::AcceptedOnL2);
+
+        for tx in latest_block.transactions {
+            assert_tx_successful(tx.transaction_hash(), &devnet.json_rpc_client).await;
+        }
+    }
+
+    async fn assert_pending_block_with_txs(devnet: &BackgroundDevnet) {
+        let pending_block = devnet.get_pending_block_with_txs().await.unwrap();
+
+        for tx in pending_block.transactions {
+            assert_tx_successful(tx.transaction_hash(), &devnet.json_rpc_client).await;
+        }
+    }
+
+    async fn assert_latest_block_with_receipts(devnet: &BackgroundDevnet, block_number: u64) {
+        let latest_block = &devnet
```

Comment:
unfortunately send_custom_rpc with `starknet_getBlockWithReceipts` is needed here, because is still not implemented in starknet-providers 0.10.0